ion/writer.go

```diff
@@ -164,6 +168,24 @@ func (w *writer) FieldName(val string) error {
 	return nil
 }
 
+// FieldNameSymbol sets the field name symbol for the next value written.
+// It may only be called while writing a struct.
+func (w *writer) FieldNameSymbol(val SymbolToken) error {
+	if !w.IsInStruct() {
+		w.err = errors.New("ion: Writer.FieldNameSymbol called when not writing a struct")
```

Comment:
errors.New? 
Is there a suitable error type in err.go we can use instead? If not can we create a new error type in err.go?

Reply:
Updated to UsageError